woocommerce/woocommerce-gateway-stripe
# Feat / Account keys UI

# Changes proposed in this Pull Request:

Issue: Add Account Keys UI to the new settings area.
Fixes #1835 

For acceptance criteria take a look at #1835 

Description: Create a pop-up modal which allows for adding Account Keys using the new react-based settings UI.
This was started before the team Heisenberg meetup and is functional but due to meetup planning responsibilities I couldn't devote the time needed to get it 100%. There are some failing tests still and it could use some optimization and polish. Marking as Draft for this to get picked up since the team will be focused as of now on WooPay.

# Testing instructions

There are three ways to open the modal:

1. Click **Edit Account Keys** within the General settings area on the settings panel of the react-based settings area.
2. Click **Enter account keys (advanced)** within the onboarding UI(this requires deleting your Stripe keys to prompt onboarding).
3. Click **Edit Account Keys** drop-down menu link within the **Account Details** area within the **Settings** panel.

Once the modal is open you should be able to switch between Test & Live mode; the keys shown there should be loaded from your saved Stripe keys. 
You should be able to save the keys and receive a toast notification (success or fail). 

There are some tests to make sure the modal opens but I think a test to ensure that upon form submission the `saveAccountKeys()` function is called with the correct arguments would be beneficial.

There is some prop drilling at the moment and perhaps an optimization could be made to create a custom react hook for opening the modal rather than passing down props:

https://github.com/woocommerce/woocommerce-gateway-stripe/compare/feat/account-keys-ui?expand=1#diff-d85d448546d4ee9c6d71de46179d526555442e9b6e0aa03f844d6bcb79b8e6a6R55
